osbs/api.py

```diff
@@ -859,16 +860,17 @@ def create_source_container_build(
             arrangement_version=arrangement_version
         )
 
+        error_messages = []
         if not sources_for_koji_build_nvr:
-            raise OsbsValidationException(
-                "required argument 'sources_for_koji_build_nvr' can't be None"
-            )
-
-        name, _, _ = sources_for_koji_build_nvr.split('-', 3)
+            error_messages.append("required argument 'sources_for_koji_build_nvr' can't be None")
+        if not component:
+            error_messages.append("required argument 'component' can't be None")
+        if error_messages:
+            raise OsbsValidationException(", ".join(error_messages))
```

Comment:
Why have the keyword arguments default to `None` in the method signature?

Reply:
mmm right we can just have it mandatory